credentials/alts/internal/handshaker/handshaker_test.go

```diff
@@ -283,3 +283,61 @@ func (s) TestPeerNotResponding(t *testing.T) {
 		t.Errorf("ClientHandshake() = %v, want %v", got, want)
 	}
 }
+
+func (s) TestNewClientHandshaker(t *testing.T) {
+	conn := testutil.NewTestConn(nil, nil)
+	clientConn := &grpc.ClientConn{}
+	opts := &ClientHandshakerOptions{}
+	hs, err := NewClientHandshaker(context.Background(), clientConn, conn, opts)
+	if err != nil {
+		t.Errorf("NextClientHandshaker returned unexpected error: %v", err)
+	}
+	altsHs := hs.(*altsHandshaker)
+	if altsHs.stream != nil {
+		t.Errorf("altsHandshaker.stream is non-nil")
+	}
+	if got, want := altsHs.conn, conn; got != want {
+		t.Errorf("altsHandshaker.conn: %v, want: %v", got, want)
+	}
+	if got, want := altsHs.clientConn, clientConn; got != want {
+		t.Errorf("altsHandshaker.clientConn: %v, want: %v", got, want)
+	}
+	if got, want := altsHs.clientOpts, opts; got != want {
+		t.Errorf("altsHandshaker.clientOpts: %v, want: %v", got, want)
+	}
+	if altsHs.serverOpts != nil {
+		t.Errorf("altsHandshaker.serverOpts is non-nil")
+	}
+	if got, want := altsHs.side, core.ClientSide; got != want {
+		t.Errorf("altsHandshaker.side: %v, want: %v", got, want)
+	}
```

Comment:
Please compare complete structs instead of individual fields. Please see: https://google.github.io/styleguide/go/decisions#full-structure-comparisons

Same applies to the below test as well.

Reply:
Did this, but needed two standalone checks for stream and clientConn, since we are not looking to do deep equality checks for those fields.